src/python/pants/backend/terraform/dependencies.py

```diff
@@ -107,66 +98,58 @@ class TerraformInitRequest:
-    address_input = request.root_module.to_address_input()
-    module_address = await Get(Address, AddressInput, address_input)
-    module = await Get(
-        WrappedTarget,
-        WrappedTargetRequest(
-            module_address, description_of_origin=address_input.description_of_origin
-        ),
+    module_dependencies = await Get(
+        TransitiveTargets, TransitiveTargetsRequest((request.dependencies.address,))
     )
 
-    root_dependencies, module_dependencies = await MultiGet(
-        Get(Targets, DependenciesRequest(request.dependencies)),
-        Get(Targets, DependenciesRequest(module.target.get(TerraformDependenciesField))),
-    )
-
-    source_files, backend_config, dependencies_files = await MultiGet(
-        Get(SourceFiles, SourceFilesRequest([module.target.get(SourcesField)])),
+    backend_config, lockfile, dependencies_files = await MultiGet(
         Get(SourceFiles, SourceFilesRequest([request.backend_config])),
         Get(
-            SourceFiles,
-            SourceFilesRequest(
-                [tgt.get(SourcesField) for tgt in (*root_dependencies, *module_dependencies)]
+            Snapshot,
+            PathGlobs(
+                [(Path(request.root_module.address.spec_path) / ".terraform.lock.hcl").as_posix()]
             ),
         ),
+        Get(
+            SourceFiles,
+            SourceFilesRequest([tgt.get(SourcesField) for tgt in module_dependencies.dependencies]),
+        ),
     )
-    files_by_directory = partition_files_by_directory(source_files.files)
-
     fetched_deps = await Get(
         TerraformDependenciesResponse,
         TerraformDependenciesRequest(
-            source_files,
-            tuple(files_by_directory.keys()),
+            request.root_module.address.spec_path,
             backend_config,
+            lockfile,
             dependencies_files,
             initialise_backend=request.initialise_backend,
+            upgrade=request.upgrade,
         ),
     )
 
-    merged_fetched_deps = await Get(Digest, MergeDigests([x[1] for x in fetched_deps.fetched_deps]))
-
-    sources_and_deps = await Get(
-        Digest,
-        MergeDigests(
-            [source_files.snapshot.digest, merged_fetched_deps, dependencies_files.snapshot.digest]
-        ),
+    all_terraform_files = await Get(
+        Digest, MergeDigests([fetched_deps.digest, dependencies_files.snapshot.digest])
     )
 
-    assert len(files_by_directory) == 1, "Multiple directories found, unable to identify a root"
-    chdir, files = next(iter(files_by_directory.items()))
-    return TerraformInitResponse(sources_and_deps, tuple(files), chdir)
+    return TerraformInitResponse(
+        terraform_dependencies=fetched_deps.digest,
+        terraform_files=dependencies_files,  # TODO: I think this includes the wrong files (all the dependencies, not just TF ones). It's now a bit muddled which files are actually TF files
```

Comment:
Yes, but maybe that is correct? Why else would there be such a dep?

Reply:
One of them would be just the terraform dependencies (like, modules and such), the other would be pants-level dependencies (including, eg, assets). I've gotta have a think about whether that distinction is useful.

Reply:
I'm pretty sure the way it is now is correct. `terraform_dependencies` gets the actual dependencies that we'd need for `export`. I'm not sure that I'd want only the dependencies that were terraform files. I think it's just that the naming is bad now. Also, maybe between `terraform_dependencies`, `terraform_files`, and `sources_and_deps`, there's a redundancy somewhere.